### ISSUE TITLE
Build failure on ghc-9.0.2

### ISSUE THREAD

**Issue by debug-ito** [andrewthad/haskell-ip#77]
I tried to build ip-1.7.3 with ghc-9.0.2 and encountered the following error.

```
src/Data/Word/Synthetic/Word12.hs:92:48: error:
    • Couldn't match expected type ‘Word#’ with actual type ‘Word’
    • In the first argument of ‘narrow12Word#’, namely
        ‘(integerToWord i)’
      In the first argument of ‘W12#’, namely
        ‘(narrow12Word# (integerToWord i))’
      In the expression: W12# (narrow12Word# (integerToWord i))
   |
92 |   fromInteger i         = W12# (narrow12Word# (integerToWord i))
   |                                                ^^^^^^^^^^^^^^^

src/Data/Word/Synthetic/Word12.hs:131:32: error:
    Variable not in scope: smallInteger :: Int# -> Integer
    |
131 |   toInteger (W12# x#)        = smallInteger (word2Int# x#)
    |                                ^^^^^^^^^^^^
```

Is the commit 7eaa44bd521ea195c5a9fe1254701de821577a8f the fix for this problem? If so, could you release it?

**Comment by andrewthad** [comment]
I've just released this at 1.7.4 to hackage. Can you confirm that it builds?

**Comment by debug-ito** [comment]
Thanks! I confirmed that it built with ghc-9.0.1
